cadence/nft-lock-box/NFTLockBox.cdc

```diff
@@ -0,0 +1,160 @@
+
+            self.publicKeys[token.id] = publicKey
+
+            collection.deposit(token: <- token)
+        }
+
+        // Claim an NFT by ID using a claim signature and
+        // deliver it to the provided address.
+        //
+        // If the NFT exists and the signature is valid,
+        // this function will deposit the NFT into the account
+        // with the provided address. 
+        //
+        // The account must expose a public NonFungibleToken.CollectionPublic
+        // capability at `self.receiverPath` that can accept NFTs of this type.
+        //
+        // A signature is generated by signing:
+        //
+        //   SHA3_256(
+        //     "FLOW-V0.0-user" + BYTES(ADDRESS) + BIG_ENDIAN_BYTES(ID)
+        //   )
+        //
+        pub fun claim(
+            id: UInt64, 
+            address: Address,
+            signature: [UInt8]
+        ) {
+            let collection = self.collection.borrow()!
+
+            // We withdraw the NFT before verifying the signature
+            // in order to fail as early as possible if the NFT
+            // does not exist in this collection.
+            let nft <- collection.withdraw(withdrawID: id)
+            let nftType = nft.getType()
+
+            let rawPublicKey = self.publicKeys.remove(key: id) 
+                ?? panic("NFT is not claimable or has already been claimed")
+
+            let publicKey = PublicKey(
+                publicKey: rawPublicKey,
+                signatureAlgorithm: SignatureAlgorithm.ECDSA_P256
+            )
+
+            let message = self.makeClaimMessage(address: address, id: id)
+
+            let isValidClaim = publicKey.verify(
+                signature: signature,
+                signedData: message,
+                domainSeparationTag: "FLOW-V0.0-user",
+                hashAlgorithm: HashAlgorithm.SHA3_256
+            )
+
+            assert(isValidClaim, message: "invalid claim signature")
+
+            let receiver = getAccount(address)
+                .getCapability(self.receiverPath)
+                .borrow<&{NonFungibleToken.CollectionPublic}>()!
+
+            receiver.deposit(token: <- nft)
+
+            emit Claimed(
+                nftType: nftType,
+                nftID: id,
+                from: self.owner?.address,
+                recipient: address
+            )
+        }
+
+        pub fun makeClaimMessage(address: Address, id: UInt64): [UInt8] {
+            let addressBytes = address.toBytes()
+            let idBytes = id.toBigEndianBytes()
+
+            return addressBytes.concat(idBytes)
+        }
+    }
+
+    pub fun createLockBox(
+        collection: Capability<&{NonFungibleToken.Provider, NonFungibleToken.Receiver}>,
+        receiverPath: PublicPath
+    ): @LockBox {
+        return <- create LockBox(collection: collection, receiverPath: receiverPath)
+    }
+
+    init() {
+        self.DefaultLockBoxStoragePath = /storage/NFTLockBox
```

Comment:
Might need to make the name of the contract and paths a little more specific to avoid conflicting with other contracts

Reply:
Yeah, I agree. Do you think it would be valid to just remove these "default" storage paths entirely and just require the caller to choose a path? Each lock box is likely bound to a specific NFT type, so I feel like it might make sense for the user to choose a path that's similar to the standard NFT collection path for their project.

e.g. `/storage/KittyItemsLockBox`

Reply:
yeah, thats a good point. Let the users make their own paths

Reply:
Removed these default storage paths in favour of user-defined paths.